tests/conftest.py

```diff
@@ -81,8 +81,10 @@ def pytest_generate_tests(metafunc):
                 callspec.marks.append(
                     pytest.mark.skip(f"Backend {backend} not available.")
                 )
-            if backend == metafunc.config.getoption("ref"):
-                # FIXME: This is skipping most tests on finufft, #140
+            if metafunc.config.getoption(
+                "backend"
+            ) != [] and backend != metafunc.config.getoption("backend"):
+                # Skip tests if the backend does not match what we want to test.
                 callspec.marks.append(
-                    pytest.mark.skip("Not testing ref backend with self.")
+                    pytest.mark.skip(f"Not testing {backend} as not requested")
                 )
```

Comment:
This is slightly more sophisticated, as we moved from skipping if we are testing with reference to
testing only if we request for it. Else, we skip. @paquiteau to comment in case this causes new test holes.

Reply:
I think this change is redundant with what is in line 20-30 above (we modify the availability of the backend dynamically in the tests config). I would rather advise to remove these lines (84-90) and see what happens.

The whole idea of this was to not test the backend against itself (it does not make sense, and it save time, esp. for the `test_interfaces.py` tests).

Reply:
Thats weird, in that case, for some reason we trigger autodiff tests for `finufft` when the requested backend is `tensorflow` :P . This is why I added it in first place..